projects/igniteui-angular/src/lib/list/list.component.spec.ts

```diff
@@ -445,6 +446,95 @@ describe('List', () => {
+
+        const firstItem = list.items[0] as IgxListItemComponent;
+        const leftPanTmpl = firstItem.leftPanningTemplateElement;
+        const rightPanTmpl = firstItem.rightPanningTemplateElement;
+        const itemNativeElements = fixture.debugElement.queryAll(By.css('igx-list-item'));
+
+        /* Click and drag item right */
+        clickAndDrag(itemNativeElements[1], 0.3);
+        fixture.detectChanges();
+
+        expect(leftPanTmpl.nativeElement.style.visibility).toBe('hidden');
+        expect(rightPanTmpl.nativeElement.style.visibility).toBe('visible');
+    });
+
+    it('checking the panLeftTemplate is not visible when releasing a list item.', fakeAsync(() => {
+        const fixture = TestBed.createComponent(ListWithPanningTemplatesComponent);
+        const list = fixture.componentInstance.list;
+        fixture.detectChanges();
+
+        const firstItem = list.items[0] as IgxListItemComponent;
+        const leftPanTmpl = firstItem.leftPanningTemplateElement;
+        const rightPanTmpl = firstItem.rightPanningTemplateElement;
+        const itemNativeElements = fixture.debugElement.queryAll(By.css('igx-list-item'));
+
+        /* Pan item left */
+        panItem(itemNativeElements[1], -0.3);
+        tick(600);
+        fixture.detectChanges();
+
+        expect(leftPanTmpl.nativeElement.style.visibility).toBe('hidden');
+        expect(rightPanTmpl.nativeElement.style.visibility).toBe('hidden');
+    }));
+
+    it('checking the panRightTemplate is not visible when releasing a list item.', fakeAsync(() => {
+        const fixture = TestBed.createComponent(ListWithPanningTemplatesComponent);
+        const list = fixture.componentInstance.list;
+        fixture.detectChanges();
+
+        const firstItem = list.items[0] as IgxListItemComponent;
+        const leftPanTmpl = firstItem.leftPanningTemplateElement;
+        const rightPanTmpl = firstItem.rightPanningTemplateElement;
+        const itemNativeElements = fixture.debugElement.queryAll(By.css('igx-list-item'));
+
+        /* Pan item right */
+        panItem(itemNativeElements[1], 0.3);
+        tick(600);
+        fixture.detectChanges();
+
+        expect(leftPanTmpl.nativeElement.style.visibility).toBe('hidden');
+        expect(rightPanTmpl.nativeElement.style.visibility).toBe('hidden');
+    }));
+
+    it('checking the header list item does not have panning and content containers.', () => {
+        const fixture = TestBed.createComponent(ListWithPanningTemplatesComponent);
+        const list = fixture.componentInstance.list;
+        fixture.detectChanges();
+
+        const headers = list.headers;
+        let header;
+        for (let i = 0; i < headers.length; i++) {
+            header = headers[i] as IgxListItemComponent;
+            expect(header.leftPanningTemplateElement).toBeUndefined();
+            expect(header.rightPanningTemplateElement).toBeUndefined();
+            expect(header.contentElement).toBe(null);
+        }
+    });
+
```

Comment:
Add tests verifying that setting the ILIstItemPanningEventArgs cancel property to true brings the item back in view.

Reply:
Requested tests have been added.